modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionServiceImpl.java

```diff
@@ -796,6 +802,9 @@ public IgniteRel visit(IgniteTableScan rel) {
                 @Override
                 public IgniteRel visit(IgniteTableModify rel) {
                     int tableId = rel.getTable().unwrap(IgniteTable.class).id();
+
+                    // TODO IGNITE-19499 Remove rewriting tableId.
+                    tableId = deps.internalTable(tableId).tableId();
                     List<NodeWithTerm> assignments = fragment.mapping().updatingTableAssignments();
 
                     enlist(tableId, assignments);
```

Comment:
assignments can be null

Reply:
Does `null` assigments means the table is not ready? 
If so, then we shouldn't get null here because we wait for table readiness before start query planning.